applications/GeoMechanicsApplication/tests/cpp_tests/test_apply_k0_procedure_process.cpp

```diff
@@ -47,75 +50,72 @@ ModelPart& PrepareTestModelPart(Model& rModel)
     return result;
 }
 
-bool AllElementsHaveSameCouplingOption(const ModelPart::ElementsContainerType& rElements,
-                                       GeoLinearElasticLaw::IsCouplingWanted ExpectedCouplingOption)
+bool ElementConsidersDiagonalEntriesOnlyAndNoShear(const Element& rElement)
 {
-    return std::all_of(rElements.begin(), rElements.end(), [ExpectedCouplingOption](const auto& rElement) {
-        auto p_constitutive_law = dynamic_cast<const GeoLinearElasticLaw*>(
-            rElement.GetProperties().GetValue(CONSTITUTIVE_LAW).get());
-        return p_constitutive_law->GetCouplingOption() == ExpectedCouplingOption;
-    });
+    auto p_constitutive_law =
+        dynamic_cast<const GeoLinearElasticLaw*>(rElement.GetProperties().GetValue(CONSTITUTIVE_LAW).get());
+    return p_constitutive_law && p_constitutive_law->GetConsiderDiagonalEntriesOnlyAndNoShear();
 }
 
 } // namespace
 
 namespace Kratos::Testing
 {
 
-KRATOS_TEST_CASE_IN_SUITE(ExpectCouplingTermsWhenForceDecoupledElasticityFlagIsNotDefined, KratosGeoMechanicsFastSuite)
+KRATOS_TEST_CASE_IN_SUITE(AllElementsConsiderDiagonalEntriesOnlyAndNoShearWhenUseStandardProcedureFlagIsNotDefined,
+                          KratosGeoMechanicsFastSuite)
 {
     Model model;
     auto& r_model_part = PrepareTestModelPart(model);
 
-    Parameters k0_settings; // 'force_decoupled_elasticity' is not defined
+    Parameters k0_settings; // 'use_standard_procedure' is not defined, assume it to be true
 
     ApplyK0ProcedureProcess process{r_model_part, k0_settings};
     process.ExecuteInitialize();
 
-    KRATOS_EXPECT_TRUE(AllElementsHaveSameCouplingOption(
-        r_model_part.Elements(), GeoLinearElasticLaw::IsCouplingWanted::Yes))
+    KRATOS_EXPECT_TRUE(boost::algorithm::all_of(r_model_part.Elements(), ElementConsidersDiagonalEntriesOnlyAndNoShear))
 }
-KRATOS_TEST_CASE_IN_SUITE(ExpectCouplingTermsWhenForceDecoupledElasticityFlagIsOff, KratosGeoMechanicsFastSuite)
+
+KRATOS_TEST_CASE_IN_SUITE(AllElementsConsiderDiagonalEntriesOnlyAndNoShearWhenUsingStandardProcedure,
+                          KratosGeoMechanicsFastSuite)
 {
     Model model;
     auto& r_model_part = PrepareTestModelPart(model);
 
-    Parameters k0_settings{R"({"force_decoupled_elasticity": false})"};
+    Parameters k0_settings{R"({"use_standard_procedure": true})"};
 
     ApplyK0ProcedureProcess process{r_model_part, k0_settings};
     process.ExecuteInitialize();
 
-    KRATOS_EXPECT_TRUE(AllElementsHaveSameCouplingOption(
-        r_model_part.Elements(), GeoLinearElasticLaw::IsCouplingWanted::Yes))
+    KRATOS_EXPECT_TRUE(boost::algorithm::all_of(r_model_part.Elements(), ElementConsidersDiagonalEntriesOnlyAndNoShear))
```

Comment:
Is there a specific reason we move to the boost version instead of the stl one with iterators? I like the readability, just wondering about stl vs 'third party' (although boost is of course a special kind of 3rd party in my opinion)

Reply:
The primary reason to use `boost::algorithm::all_of` is it provides an overload that accepts a range (which I feel is easier to read than a pair of iterators; also, you are very explicit that you want to visit all elements in the container, and you can't accidentally pass a mismatching pair of iterators). As of C++20, ranges are available in the standard. Since we are still at C++17, this is a nice alternative. And since it doesn't pull in a new dependency, I feel this is acceptable. We may consider to use it in other places, too.